lib/dynflow/persistence_adapters/sequel.rb

```diff
@@ -67,17 +79,61 @@ def save_action(execution_plan_id, action_id, value)
         save :action, { execution_plan_uuid: execution_plan_id, id: action_id }, value
       end
 
+      def save_envelope(data)
+        save :envelope, {}, data
+      end
+
+      def pull_envelopes(receiver_id)
+        db.transaction do
+          data_set = table(:envelope).where(receiver_id: receiver_id).to_a
+
+          envelopes = data_set.map do |record|
+            Serializable::AlgebrickSerializer.instance.load(record[:data], Dispatcher::Envelope)
+          end
+
+          table(:envelope).where(id: data_set.map { |d| d[:id] }).delete
+          return envelopes
+        end
+      end
+
+      def push_envelope(envelope)
+        table(:envelope).insert(prepare_record(:envelope, envelope))
+      end
+
+      def insert_coordinator_record(value)
+        save :coordinator_record, {}, value
+      end
```

Comment:
This may not be the best place for these methods since the Sequel adapter may be used for persistence but different adapter for coordinator.